src/ResizeSensor.js

```diff
@@ -163,19 +175,23 @@
             attachResizeEvent(element, callback);
         }
 
-        this.detach = function() {
+        this.detach = function(ev) {
             if (isCollectionTyped) {
                 var i = 0, j = element.length;
                 for (; i < j; i++) {
-                    ResizeSensor.detach(element[i]);
+                    ResizeSensor.detach(element[i], ev);
                 }
             } else {
-                ResizeSensor.detach(element);
+                ResizeSensor.detach(element, ev);
             }
         };
     };
 
-    ResizeSensor.detach = function(element) {
+    ResizeSensor.detach = function(element, ev) {
+        if(typeof ev == "function"){
+            element.resizedAttached.remove(ev);
+            if(element.resizedAttached.length()) return;
+        }
```

Comment:
This makes the following possible:

```
var fn1 = function(){ console.log("resize 1") }
ResizeSensor(element, fn1);

var fn2 = function(){ console.log("resize 2") }
ResizeSensor(element, fn2);

// at this point, resizing el would log
//
// resize 1
// resize 2
// resize 1
// resize 2
//...

ResizeSensor.detach(element, fn2);

// now, resizing would log only
//
// resize 1
// resize 1
// resize 1
//...
```

This sort of approach mirrors the native addEventListener/removeEventListener interface, which allows you to remove specific events by passing a ref to the original function.

In the event that you call `ResizeSensor.detach(element,fn)` and the fn removed leaves none left in the event queue, we do the rest of the clean up as well, else we leave the rest since it's necessary for the remaining events.